doc: llext: improve documentation for the llext subsystem (#75126)
zephyrproject-rtos/zephyr: doc/services/llext/index.rst

Linkable Loadable Extensions (LLEXT)
####################################

.. toctree::
   :maxdepth: 1

   config
   build
   load
   api

The LLEXT subsystem provides a toolbox for extending the functionality of an
application at runtime with linkable loadable code.

> Extensions can be loaded from precompiled ELF formatted data which is

Reviewer: ```suggestion
Extensions are precompiled executables in ELF format that can be
```
(`precompiled ELF formatted data` is an obscure formulation IMHO)
